terraform/context_components.go

```diff
@@ -16,7 +15,7 @@ type contextComponentFactory interface {
 	// ResourceProvider creates a new ResourceProvider with the given
 	// type. The "uid" is a unique identifier for this provider being
 	// initialized that can be used for internal tracking.
-	ResourceProvider(typ, uid string) (providers.Interface, error)
+	ResourceProvider(fqn, uid string) (providers.Interface, error)
```

Comment:
Could we make sure we got this right in all callers by also changing this argument to be of type `addrs.Provider`?

I'm hesitant to add a "while we're here" to an already-large changeset, so please do feel free to ignore this for the moment and we can revisit later, but I also wanted to note that removing this `uid string` argument has been a low-priority cleanup for a while now; AFAIK we no longer use it for anything, so we could potentially drop it while we're breaking the signature of this function anyway.